stl/inc/mutex

```diff
@@ -544,23 +545,47 @@ int __stdcall _Callback_once(void*, void* _Pv, void**) { // adapt call_once() to
 
 template <class _Fn, class... _Args>
 void(call_once)(once_flag& _Flag, _Fn&& _Fx, _Args&&... _Ax) { // call _Fx(_Ax...) once
-    using _Tuple = tuple<_Fn&&, _Args&&..., exception_ptr&>;
-    using _Seq   = make_index_sequence<1 + sizeof...(_Args)>;
+    int pending;
+    bool fallback = false;
+    if (!__std_execute_once_begin(_Flag, pending, fallback)) {
+        if (fallback) {
+            using _Tuple = tuple<_Fn&&, _Args&&..., exception_ptr&>;
+            using _Seq   = make_index_sequence<1 + sizeof...(_Args)>;
 
-    exception_ptr _Exc;
-    _Tuple _Tup(_STD forward<_Fn>(_Fx), _STD forward<_Args>(_Ax)..., _Exc);
+            exception_ptr _Exc;
+            _Tuple _Tup(_STD forward<_Fn>(_Fx), _STD forward<_Args>(_Ax)..., _Exc);
 
-    _Execute_once_fp_t _Fp = &_Callback_once<_Tuple, _Seq, 1 + sizeof...(_Args)>;
+            _Execute_once_fp_t _Fp = &_Callback_once<_Tuple, _Seq, 1 + sizeof...(_Args)>;
 
-    if (_Execute_once(_Flag, _Fp, _STD addressof(_Tup)) != 0) {
-        return;
-    }
+            if (_Execute_once(_Flag, _Fp, _STD addressof(_Tup)) != 0) {
+                return;
+            }
 
-    if (_Exc) {
-        _STD rethrow_exception(_Exc);
+            if (_Exc) {
+                _STD rethrow_exception(_Exc);
+            }
+        }
+        _XGetLastError();
+    }
+    if (!pending) {
+        return; // already initialized (won the race otherwise)
     }
 
-    _XGetLastError();
+    struct _Execute_once_scope_t {
+        ~_Execute_once_scope_t() {
+            if (!__std_execute_once_complete(_Once_flag, _Flags)) {
+                terminate(); // Cannot strongly-exception-safely throw,
+                // callback already executed (and might have thrown itself)
+            }
+        }
+
+        once_flag& _Once_flag;
+        unsigned long _Flags;
+    } _Execute_once_scope = {_Flag, 0x4 /*INIT_ONCE_INIT_FAILED*/};
+
+    _STD invoke(_STD forward<_Fn>(_Fx), _STD forward<_Args>(_Ax)...);
+
+    _Execute_once_scope._Flags = 0; // succeeded

```

Comment:
I find it mildly distressing that we have `_Flag` and `_Execute_once_scope._Flags`. (Similarly-named variables create the potential for confusion.) They're not shadowing and they have different types, so no changes requested.

Reply:
I've renamed it in parameter from `_Flag` to `_Once_flag`, probably should be done here as well.

Reply:
Got away by making this parameter just `_Succeeded` and converting it into flags directly in API  call expression.